analytics-example-widget/scripts/components/ChartComponent.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+
+import * as Chart_Contracts from "Charts/Contracts";
+import * as Chart_Service from "Charts/Services";
+
+/**
+ * Stateless parameters (creation-time) for constructing a ChartComponent.
+ */
+export class ChartComponentProps {
+    chartOptions: Chart_Contracts.CommonChartOptions
+}
+
+/**
+ * 
+ */
+export class ChartComponent extends React.Component<ChartComponentProps, any> {    
+    //TODO: This is a temporary workaround to compensate for lack of dispose handler on Chart object, which is resolved with M126 binaries.
+    //  After chart + dispose API exposed in updated public API, code associated with this can be removed.
+    
+    private isMounted:boolean=false;    
+
+    public render() {
+        return (
+            <div className="chart-component"></div>
+        );
+    }
+
+    public componentDidMount(): void {
+        this.isMounted=true;
+        this.ensureChartIsInstantiated(this.props.chartOptions);
+
+    }
+
+    public componentWillUnmount(): void {        
+        this.ensureChartIsDisposed();
+        this.isMounted=false;
+    }
+
+    public componentDidUpdate(): void {
+        this.ensureChartIsInstantiated(this.props.chartOptions);
+    }
+
+    private ensureChartIsInstantiated(chartOptions: Chart_Contracts.CommonChartOptions) {
+        Chart_Service.ChartsService.getService().then((chartService)=>{
+            this.ensureChartIsDisposed();
+
+            if (this.isMounted) {
+                let container = ReactDOM.findDOMNode(this);
+                let $wrappedContainer = $(container);
+
+                if (chartOptions) {            
+                    chartOptions = this.updateChartOptions(chartOptions);
+                    chartService.createChart($wrappedContainer, chartOptions);            
+                }
+            }
+        });
+    }
+
+    private ensureChartIsDisposed(): void {
+        //TODO: Use Chart control dispose API, once it is available on public contract.
+        if(this.isMounted){
+            $(ReactDOM.findDOMNode(this)).empty();        
```

Comment:
i think you meant to do unmountMode

Reply:
Intent here is to clear out the content of the node - Problem was async nature of rendering chart after the normal render flow, due to chart service, and need for a parent dom element.

The chart component section will be updated in a couple days after a prod bugfix rolls out, and ultimately moved to be provided from charting API in future sprint.